lib/include/jxl/decode.h

```diff
@@ -913,6 +953,144 @@ JXL_EXPORT JxlDecoderStatus JxlDecoderSetJPEGBuffer(JxlDecoder* dec,
+ * characters plus a null termination character. In case of a compressed box,
+ * this returns the type of the underlying uncompressed box, not "brob". Use
+ * @see JxlDecoderGetBoxTypeRaw to get the raw type.
+ *
+ * @param dec decoder object
+ * @param type buffer to copy the type into
+ * @return JXL_DEC_SUCCESS if the value is available, JXL_DEC_ERROR if not, for
+ *    example the JXL file does not use the container format.
+ */
+JXL_EXPORT JxlDecoderStatus JxlDecoderGetBoxType(JxlDecoder* dec, char type[5]);
+
+/**
+ * Outputs whether the given type equals the type returned by
+ * JxlDecoderGetBoxType.
+ *
+ * @param dec decoder object
+ * @param type box type to compare with, e.g. "Exif". Typenames are 4 non-null
+ * characters, passing a const char* string with null terminator is possible but
+ * not required.
+ * @return JXL_TRUE if the box type is available and equal to the given type,
+ * JXL_FALSE otherwise.
+ */
+JXL_EXPORT JXL_BOOL JxlDecoderBoxTypeEquals(JxlDecoder* dec,
+                                            const char type[4]);
+
+/**
+ * Outputs the type of the current box, after a JXL_DEC_BOX event occured, as 4
+ * characters plus a null termination character. If the box is brotli
+ * compressed, it returns the raw type of the box, which is "brob", rather than
+ * the underlying type. Use @see JxlDecoderGetBoxType to get the underlying
+ * type.
+ *
+ * @param dec decoder object
+ * @param type buffer to copy the type into
+ * @return JXL_DEC_SUCCESS if the value is available, JXL_DEC_ERROR if not, for
+ *    example the JXL file does not use the container format.
+ */
+JXL_EXPORT JxlDecoderStatus JxlDecoderGetBoxTypeRaw(JxlDecoder* dec,
+                                                    char type[5]);
+
+/**
+ * Outputs whether the given type equals the type returned by
+ * JxlDecoderGetBoxTypeRaw.
+ *
+ * @param dec decoder object
+ * @param type box type to compare with, e.g. "Exif". Typenames are 4 non-null
+ * characters, passing a const char* string with null terminator is possible but
+ * not required.
+ * @return JXL_TRUE if the raw box type is available and equal to the given
+ * type, JXL_FALSE otherwise.
+ */
+JXL_EXPORT JXL_BOOL JxlDecoderBoxTypeRawEquals(JxlDecoder* dec,
+                                               const char type[4]);
+
+/**
+ * Outputs whether the box is a brotli compressed box. This implies its raw
+ * type is "brob". Its underlying type is still returned with
+ * JxlDecoderGetBoxType. The data returned to the box buffer is transparently
+ * decompressed, unless JxlDecoderSetBoxModeRaw is enabled, so this function is
+ * not needed to just get the data, as long as the library is compiled with
+ * brotli.
+ *
+ * @param dec decoder object
+ * @return JXL_TRUE if the current box is a compressed box, false if it's an
+ * uncompressed box, or there's no box such as in case the container format is
+ * not used.
+ */
+JXL_EXPORT JXL_BOOL JxlDecoderIsBoxCompressed(JxlDecoder* dec);
```

Comment:
Is this equivalent to BoxTypeRaw == "brob", right? Not sure this deserves its own function.

Reply:
This redundancy too was pointed out by two people, so I'll remove it too.

Note that handling type == "brob" requires specification knowledge, so in a sense the user is writing the parser, while JxlDecoderIsBoxCompressed would wrap this